jmclient/test/test_wallet_rpc.py

```diff
@@ -712,6 +718,88 @@ def process_get_seed_response(self, response, code):
         json_body = json.loads(response.decode('utf-8'))
         assert json_body["seedphrase"]
 
+class TrialTestWRPC_JWT(WalletRPCTestBase, unittest.TestCase):
+    @defer.inlineCallbacks
+    def test_refresh_token(self):
+        """Test refresh token endpoint
+
+        1. Valid access, valid refresh -> Success
+        2. Valid access, invalid refresh -> Failed
+        3. Valid access, expired refresh -> Failed
+        4. Expired access, valid refresh -> Success
+        5. Expired access, invalid refresh -> Failed
+        6. Expired access, expired refresh -> Failed
+        7. Invalid access, valid refresh -> Failed
+        """
+
+        agent = get_nontor_agent()
+        addr = self.get_route_root()
+        addr += "/token/refresh"
+        addr = addr.encode()
+
+        for access_token, refresh_token, is_successful_response in [
+            ("valid", "valid", True),
+            ("valid", "invalid", False),
+            ("valid", "expired", False),
+            ("expired", "valid", True),
+            ("expired", "invalid", False),
+            ("expired", "expired", False),
+            ("invalid", "valid", False),
+        ]:
+            access_token, refresh_token, is_successful_response = "valid", "valid", True
```

Comment:
@roshii I don't understand: this line is inside the `for` loop, but it resets the values of the loop variables so they're always the same?

Reply:
> this line is inside the `for` loop, but it resets the values of the loop variables so they're always the same?

I can remember creating the test without the `for` loop with this line, should be gone indeed...